src/Magnum/Octree/Octree.hpp

```diff
@@ -0,0 +1,344 @@
+
+    return *this;
+}
+
+template <typename T>
+Int Octree<T>::cubeInFrustum(const Vector4 (&frustum)[6]) {
+    Int cornersInsideFrustumCount = 0;
+    Int frustumPlanesCount = 0;
+
+    //if distance from a point to the plane is positive, the point lies in front of the plane (inside the frustum)
+    //distance = A * X + B * Y + C * Z + D
+
+    //we test for each corner of an octant, if it is inside the frustum.
+    //If only a part of the corners is inside, the octant has to be looked at more precisely
+
+    for(int i = 0; i < 6; ++i)
+    {
+        const Vector4 f = frustum[i];
+        cornersInsideFrustumCount = 0;
+        if(f.x() * (center().x() - radius()) + f.y() * (center().y() - radius()) + f.z() * (center().z() - radius()) + f.w() > 0) {
+            ++cornersInsideFrustumCount;
+        }
+        if(f.x() * (center().x() + radius()) + f.y() * (center().y() - radius()) + f.z() * (center().z() - radius()) + f.w() > 0) {
+            ++cornersInsideFrustumCount;
+        }
+        if(f.x() * (center().x() - radius()) + f.y() * (center().y() + radius()) + f.z() * (center().z() - radius()) + f.w() > 0) {
+            ++cornersInsideFrustumCount;
+        }
+        if(f.x() * (center().x() + radius()) + f.y() * (center().y() + radius()) + f.z() * (center().z() - radius()) + f.w() > 0) {
+            ++cornersInsideFrustumCount;
+        }
+        if(f.x() * (center().x() - radius()) + f.y() * (center().y() - radius()) + f.z() * (center().z() + radius()) + f.w() > 0) {
+            ++cornersInsideFrustumCount;
+        }
+        if(f.x() * (center().x() + radius()) + f.y() * (center().y() - radius()) + f.z() * (center().z() + radius()) + f.w() > 0) {
+            ++cornersInsideFrustumCount;
+        }
+        if(f.x() * (center().x() - radius()) + f.y() * (center().y() + radius()) + f.z() * (center().z() + radius()) + f.w() > 0) {
+            ++cornersInsideFrustumCount;
+        }
+        if(f.x() * (center().x() + radius()) + f.y() * (center().y() + radius()) + f.z() * (center().z() + radius()) + f.w() > 0) {
+            ++cornersInsideFrustumCount;
+        }
+        if(cornersInsideFrustumCount == 0) {
+            return 0;
+        }
+        if(cornersInsideFrustumCount == 8) {
+            ++frustumPlanesCount;
+        }
+    }
+    return (frustumPlanesCount == 6) ? 2 : 1;
+}
+
+
+template <typename T>
+bool Octree<T>::boxInFrustum(const Shapes::AxisAlignedBox3D& box, const Vector4 (&frustum)[6]) {
+
+    /* create the 6 other vertices of the box from the 2 given vertices min, and max */
+    const Vector3 v[8] = {
+        box.min(),
+        {box.min().x(), box.min().y(), box.max().z()},
+        {box.min().x(), box.max().y(), box.min().z()},
+        {box.max().x(), box.min().y(), box.min().z()},
+        {box.min().x(), box.max().y(), box.max().z()},
+        {box.max().x(), box.min().y(), box.max().z()},
+        {box.max().x(), box.max().y(), box.min().z()},
+        box.max()};
```

Comment:
Like suggested by @Squareys it can be rewritten as

    Vector3 v[8];
    for(int i = 0; i < 8; ++i) {
         v[i] = Vector3 {
            (i & 4 ? box.max().x() : box.min().x()),
            (i & 2 ? box.max().y() : box.min().y()),
            (i & 1 ? box.max().z() : box.min().z())};
    }

respectively the above `if` blocks replaced by something like:

    for(int i = 0; i < 8; ++i) {
        xCoord = i & 4 ? (center().x() + radius()) : (center().x() - radius());
        yCoord = i & 2 ? (center().y() + radius()) : (center().y() - radius());
        zCoord = i & 1 ? (center().z() + radius()) : (center().z() - radius());

        if(f.x() * xCoord + f.y() * yCoord + f.z() * zCoord + f.w() > 0) {
            ++cornersInsideFrustumCount;
        }
    }

Or even simpler, possibly be replaced with `Math::lerp` like suggested